prov/verbs/src/verbs_eq.c

```diff
@@ -838,10 +849,8 @@ int fi_ibv_eq_open(struct fid_fabric *fabric, struct fi_eq_attr *attr,
 	dlistfd_head_free(&_eq->list_head);
 err1:
 	fastlock_destroy(&_eq->lock);
-	if (fi_ibv_using_xrc()) {
-		fastlock_destroy(&_eq->xrc.idx_lock);
-		free(_eq->xrc.conn_key_map);
-	}
+	fastlock_destroy(&_eq->xrc.idx_lock);
```

Comment:
I might have asked this before, but is a separate idx_lock needed, or could the eq lock be used?

Reply:
There is no reason that it has to be separate. Since there is 1 EQ per RDM endpoint it seemed that a more granular lock was appropriate/safer, but after looking at eq::lock it should be fine to share the lock if you prefer.

Reply:
Thanks. I have updated to use the eq::lock here.